src/test/scala/chiselTests/experimental/AutoTypenameSpec.scala

```diff
@@ -0,0 +1,75 @@
+package chiselTests
+package experimental
+
+import chisel3._
+import chisel3.experimental.HasAutoTypename
+import chisel3.util.{Decoupled, Queue}
+import circt.stage.ChiselStage
+
+class Top(gen: Bundle) extends Module {
+  val in = IO(Input(gen))
+  val out = IO(Output(gen))
+  override def desiredName = s"Top_${out.typeName}"
+}
+
+class SimpleBundle(width: Int) extends Bundle with HasAutoTypename {
+  val foo = UInt(width.W)
+}
+
+class StringlyBundle(width: Int, desc: String) extends Bundle with HasAutoTypename {
+  val foo = UInt(width.W)
+}
+
+class DataBundle(data: Data) extends Bundle with HasAutoTypename {
+  val foo = data
+}
+
+class BundleBundle(gen: HasAutoTypename) extends Bundle with HasAutoTypename {
+  val foo = gen
+}
```

Comment:
Please add a test for a Bundle with multiple parameter lists, including implicit eg.
```
class MultipleParameterLists(x: Int)(y: List[Int])(implicit z: String)`